content/tutorial-stochastic-simulations.md

```diff
@@ -0,0 +1,483 @@
+    
+    return pi
+```
+
+We can then call this function with 100 dots generated.
+Because we are using a random number generator, every time we call this function,
+we likley get a different value for $\pi$.
+
+```{code-cell} ipython3
+estimate_pi(100)
+```
+
+```{code-cell} ipython3
+estimate_pi(100)
+```
+
+We can also generate a value for $\pi$ using the `np.pi` to compare our answer to.
+
+```{code-cell} ipython3
+np.pi
+```
+
+Odds are that the values calculated above for $\pi$ above is close but not
+exactly 3.14.
+If we increase the number of dots, the value for $\pi$ will on average become
+closer to the true value.
+While it may be tempting to just call the function with an extremely large number
+of dots, this can take substatially longer to computer.
+
+```{code-cell} ipython3
+estimate_pi(1E8)
+```
+
+## Simulating radioactive decay
+
+As a final example, we will simulate the radioactive decay of the radioactive
+nuclide $^{137m}$Ba to determine the quantity of undecayed material versus time.
+We will then compare it to an analytical model to see how well the two agree.
+Radioactive nuclei decay by first-order kinetics, which means that each radioactive
+nucleus has a fixed probability of decaying at any given time.
+This is analogous to how each six-sided die in a bucket has a fixed probability
+of rolling a one each time they are dumped out.
+This analogy is so good that we can simulate the decay process by generating a
+large number of random integers with fixed probabilities.
+
+The first thing we need to is to know the probability of a nucleus decaying in
+a single second.
+Because this is a first-order process, the rate (-dA/dt) is described by
+
+$$ \frac{-dA}{dt} = kA $$
+
+where A is the quantity of $^{137m}$Ba, $t$ is time in seconds, and $k$ is the
+rate constant for this process.
+We also know that the half-life (t$_{1/2}$) of  $^{137m}$Ba is about 151 seconds
+which is related to the rate constant by the following equation.
+
+$$ t_{1/2} = \frac{ln(2)}{k} $$
+
+This means that $k$ = 4.59e$^{-3}$ s$^{-1}$.
+For first-order kinetics, this means that 4.59e$^{-3}$ fraction of the nuclei
+decays, so this is the probability we need to simulate this process using random
+numbers.
+The function below calculates the fraction of radioactive material remaining
+based on a rate constant `k` and allows the users to choose the number of
+simulated nuclei `n` and duraction of the simulation `n_final`.
+
+```{code-cell} ipython3
+def decay(k, n=1000, t_final=1000):
+    '''floats and integers -> ndarray of integers
+
+    Calculates the fraction of nuclei remaining at each second in the simulation
+    given n initial nuclei with a rate constant k of decaying each second and
+    a final simulation time t_final.
+    '''
+
+    n_undecayed = n
+    undecayed_array = np.full(t_final + 1, n)
+    
+    for second in range(1, t_final + 1):
+        decays = rng.binomial(1, p=k, size=n_undecayed).sum()
```

Comment:
What exactly did you want to show here? `decays` is binomial-distributed, but the idiomatic way to compute this with the `binomial()` method would be:

```python
decays = rng.binomial(n_undecayed, p=k)
```

If you wanted to show how the binomial is constructed out of a sum of Bernoulli trials, you can do that, but I think it's confusing to use `binomial()` to make Bernoulli trials only to sum them up. The idiom for getting Bernoulli trials with a probability of `k`, and then summing up the successes looks like this:

```python
decays = (rng.random(n_undecayed) < k).sum()
```